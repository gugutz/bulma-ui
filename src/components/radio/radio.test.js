import React from 'react'
import renderer from 'react-test-renderer'
import Radio from './radio'

describe('<Radio /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<Radio name="something">Remember me</Radio>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support disabled prop', () => {
    const tree = renderer
      .create(
        <Radio name="something" disabled>
          Remember me
        </Radio>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support checked prop', () => {
    const tree = renderer
      .create(
        <Radio name="something" checked>
          Remember me
        </Radio>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
