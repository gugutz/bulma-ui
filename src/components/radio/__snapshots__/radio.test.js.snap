// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radio /> component should render correctly 1`] = `
<label
  className="radio"
  disabled={false}
>
  <input
    checked={false}
    disabled={false}
    name="something"
    type="radio"
  />
  Remember me
</label>
`;

exports[`<Radio /> component should support checked prop 1`] = `
<label
  className="radio"
  disabled={false}
>
  <input
    checked={true}
    disabled={false}
    name="something"
    type="radio"
  />
  Remember me
</label>
`;

exports[`<Radio /> component should support disabled prop 1`] = `
<label
  className="radio"
  disabled={true}
>
  <input
    checked={false}
    disabled={true}
    name="something"
    type="radio"
  />
  Remember me
</label>
`;
